source/dem/periodic_boundaries_manipulator.cc

```diff
@@ -0,0 +1,194 @@
+  boundary_information.point_on_face = quad_point;
+}
+
+template <int dim>
+void
+PeriodicBoundariesManipulator<dim>::map_periodic_cells(
+  const parallel::distributed::Triangulation<dim> &triangulation)
+{
+  periodic_boundary_cells_information.clear();
+
+  // Iterating over the active cells in the trangulation
+  for (const auto &cell : triangulation.active_cell_iterators())
+    {
+      if (cell->is_locally_owned())
+        {
+          // Iterating over cell faces
+          for (const auto &face : cell->face_iterators())
+            {
+              // Check if face is on the periodic boundary flaged as outlet.
+              // Pairs of periodic cells are stored once.
+              if (face->boundary_id() == outlet_boundary_id)
+                {
+                  unsigned int face_id = cell->face_iterator_to_index(face);
+
+                  // Save boundaries information related to the cell on
+                  // outlet boundary.
+                  boundary_cells_info_struct<dim> boundary_information;
+                  get_boundary_info(cell, face_id, boundary_information);
+
+                  // Save boundaries information related to the cell on
+                  // Periodic boundary.
+                  boundary_cells_info_struct<dim> periodic_boundary_information;
+                  typename Triangulation<dim>::active_cell_iterator
+                    periodic_cell = cell->periodic_neighbor(face_id);
+                  get_boundary_info(periodic_cell,
+                                    cell->periodic_neighbor_face_no(face_id),
+                                    periodic_boundary_information);
+
+                  // Store both cell information in map with cell id at
+                  // outlet as key
+                  periodic_boundary_cells_information.insert(
+                    {boundary_information.cell->global_active_cell_index(),
+                     std::make_pair(boundary_information,
+                                    periodic_boundary_information)});
+                }
+            }
+        }
+    }
+}
+
+template <int dim>
+void
+PeriodicBoundariesManipulator<dim>::check_and_move_particles(
+  boundary_cells_info_struct<dim> &cell_1,
+  boundary_cells_info_struct<dim> &cell_2,
+  typename Particles::ParticleHandler<dim>::particle_iterator_range
+    &particles_in_cell)
+{
+  for (auto particle = particles_in_cell.begin();
+       particle != particles_in_cell.end();
+       ++particle)
+    {
+      Point<dim, double> particle_position = particle->get_location();
+
+      // Calculate distance between particle position and the face of
+      // boundary cell d = n•(pt_particle - pt_face)
+      double distance_with_face =
+        scalar_product(particle_position - cell_1.point_on_face,
+                       cell_1.normal_vector);
+
+
+      // If distance >= 0, particle is outside of cell (or on face).
+      // If so, particle location is modified to get moved in the periodic cell.
```

Comment:
```suggestion
      // If so, particle location is modified to get moved into the periodic cell.
```
I am not sure what do you mean here. Are you moving the particle from one cell into the periodic cell? If so this need to be into